R/connect_email.R

```diff
@@ -0,0 +1,171 @@
+
+      if (isTRUE(preview)) {
+
+        html_file <- paste0(
+          knitr_sidecar_prefix(default = "blastula"),
+          "_email.html"
+        )
+
+        html <- email$html_html
+
+        msg <- create_rmd_preview_message(subject = subject)
+
+        html <-
+          sub(
+            "(<body(?!\\w)[^>]*>)", paste0("\\1", msg),
+            html, perl = TRUE, ignore.case = TRUE
+          )
+
+        writeLines(text = html, con = html_file)
+        utils::browseURL(url = html_file)
+      }
+    }
+
+    # Set the RStudio Connect output metadata options for the email message
+    # body and for the images therein
+    rmarkdown::output_metadata$set(rsc_email_body_html = email$html_str)
+    rmarkdown::output_metadata$set(rsc_email_images = email$images)
+  }
+
+  if (!is.null(text)) {
+
+    text <- paste(text, collapse = "\n")
+
+    rmarkdown::output_metadata$set(rsc_email_body_text = text)
+  }
+
+  # Set the email subject string if this is provided
+  if (!is.null(subject)) {
+    rmarkdown::output_metadata$set(rsc_email_subject = subject)
+  }
+
+  # Attach files via the `rsc_email_attachments` parameter
+  if (!is.null(attachments)) {
+    rmarkdown::output_metadata$set(rsc_email_attachments = attachments)
+  }
+
+  # Set the `rsc_email_suppress_report_attachment` parameter to
+  # `TRUE` if we elect to exclude the default Connect attachment
+  # of the rendered .Rmd
+  if (isTRUE(attach_output)) {
+    rmarkdown::output_metadata$set(rsc_email_suppress_report_attachment = FALSE)
+  } else {
+    rmarkdown::output_metadata$set(rsc_email_suppress_report_attachment = TRUE)
+  }
+
+  invisible()
+}
+
+#' Suppress any scheduled emailing in RStudio Connect
+#'
+#' This function is useful for suppressing the scheduled emailing of a published
+#' R Markdown document. It can be invoked anywhere in the R Markdown document
+#' and is useful in a conditional statement, where the result of the condition
+#' determines whether or not email suppression should occur.
+#'
+#' Since this function needs to be invoked within an R Markdown document, the
+#' chunk option `echo=FALSE` is useful here (so that viewers of the rendered
+#' document don't have to unnecessarily read code related to email suppression).
+#' While the output is invisible, any errors related to the use of this function
+#' will be visible to the author.
+#'
+#' @param suppress A logical value for whether email suppression should occur
+#'   after publication. By default, this is `TRUE`.
+#' @export
+suppress_scheduled_email <- function(suppress = TRUE) {
+
+  rmarkdown::output_metadata$set(rsc_email_suppress_scheduled = suppress)
+}
+
+#' Utility function to generate the preview message for a Connect Email
+#'
+#' @noRd
+create_rmd_preview_message <- function(subject = NULL) {
+
+  if (!is.null(subject)) {
+    subject_ln <-
+      paste0(
+        "<br><br><strong><span style=\"font-variant: small-caps;\">",
+        "email subject: </span></strong>", subject, "<br>"
```

Comment:
The `subject` value needs to be `htmltools::htmlEscape()` escaped.

Reply:
Now done. Thanks!